### PR TITLE
DWARF5 line header parsing

### PR DESCRIPTION
**Pull request by MewtR**
- [ ] Mark this if you consider it ready to merge
- [ ] I've added tests (optional)
- [ ] I wrote some lines in the [book](https://github.com/radareorg/radare2book) (optional)

**Description**

My attempt to support DWARF5 line header parsing. Mostly done, just need to update the tests.

### CODE REVIEW

**Comment by trufae** [comment]
Ready to review now? I didnt expect that pr and thats beautiful! 🥰

**Comment by MewtR** [comment]
Thanks, yeah I think now it's fine. Let me know what you think.

**Comment by trufae** [comment]
Fix those warnings
Screenshot 2023-01-16 at 00 43 13

File: libr/bin/dwarf.c
Comment (trufae) [inline libr/bin/dwarf.c:564]: function signatures must be always 1 line for grepping purposes

File: libr/bin/dwarf.c
Comment (trufae) [inline libr/bin/dwarf.c:571]:
```suggestion
```

File: libr/bin/dwarf.c
Comment (trufae) [inline libr/bin/dwarf.c:593]: check if calloc fails

File: libr/bin/dwarf.c
Comment (trufae) [inline libr/bin/dwarf.c:565]:
```suggestion
	const size_t maxlen = 0xfff;
```

File: libr/bin/dwarf.c
Comment (trufae) [inline libr/bin/dwarf.c:633]: handle this case to avoid a crash.

File: libr/bin/dwarf.c
Comment (trufae) [inline libr/bin/dwarf.c:694]: both cases use if (mode == R_MODE_PRINT. just group that like this:

```c
if (mode == R_MODE_PRINT) {
  switch (i) {
 case DIRECTORIES: 
  break;
case FILES:
 break;
}
```

**Review by trufae — CHANGES_REQUESTED** [review]
Will be good to do some fuzzing before merging this. I'm fine to merge it like this (but after 5.8.2) or do some fuzzing in `test/fuzz/` (see the readme to setup the thing). But the patch looks quite smaller than I expected. After addressing the comments i'll do another review. I can do some fuzzing on it if i find some time this week too. But having dwarf5 support was kind of high in the wishlist.

**Comment by MewtR** [comment]
I'll look into the fuzzing a little later this week.

**Comment by trufae** [comment]
Let's merge it! i will have some time today to fuzz it and test it :) so it will fit into the release! thank you!

**Pull request merged**
